posthog/management/commands/resync_persons_on_events.py

```diff
@@ -0,0 +1,131 @@
+        """,
+        f"""
+            CREATE TABLE {mig_0006_mod.TEMPORARY_PDI2_TABLE_NAME} {{on_cluster_clause}} AS {settings.CLICKHOUSE_DATABASE}.person_distinct_id2
+            ENGINE = ReplacingMergeTree(version)
+            ORDER BY (team_id, distinct_id)
+            SETTINGS index_granularity = 128
+        """,
+        f"""
+            CREATE TABLE {mig_0006_mod.TEMPORARY_GROUPS_TABLE_NAME} {{on_cluster_clause}} AS {settings.CLICKHOUSE_DATABASE}.groups
+            ENGINE = ReplacingMergeTree(_timestamp)
+            ORDER BY (team_id, group_type_index, group_key)
+            SETTINGS index_granularity = 128 {mig_0006_mod.STORAGE_POLICY_SETTING()}
+        """,
+        # fill them querying only for the selected team_id
+        f"""
+            INSERT INTO {mig_0006_mod.TEMPORARY_PERSONS_TABLE_NAME}(team_id, id, created_at, properties, is_deleted, version)
+            SELECT
+                team_id,
+                id,
+                created_at,
+                properties,
+                0 as is_deleted,
+                ver as version
+            FROM (
+                SELECT
+                    any(team_id) as team_id,
+                    id,
+                    argMax(created_at, version) as created_at,
+                    argMax(properties, version) as properties,
+                    max(version) as ver
+                FROM
+                    person
+                WHERE
+                    person.team_id = {TEAM_ID}
+                GROUP BY
+                    id
+                HAVING
+                    max(is_deleted) = 0
+            )
+        """,
+        f"""
+            INSERT INTO {mig_0006_mod.TEMPORARY_PDI2_TABLE_NAME}(team_id, distinct_id, person_id, is_deleted, version)
+            SELECT
+                team_id,
+                distinct_id,
+                person_id,
+                0 as is_deleted,
+                ver as version
+            FROM (
+                SELECT
+                    distinct_id,
+                    argMax(person_id, version) as person_id,
+                    any(team_id) as team_id,
+                    max(version) as ver
+                FROM
+                    person_distinct_id2
+                WHERE
+                    person_distinct_id2.team_id = {TEAM_ID}
+                GROUP BY
+                    distinct_id
+                HAVING
+                    max(is_deleted) = 0
+            )
+        """,
+        # TODO: groups
+    ]
+
+    for id, step in enumerate(prep_steps):
+        run_step(id, step)
+    logger.info("Creating dictionaries")
+    mig_0006._create_dictionaries(QUERY_ID)
+    run_step(
+        f"final alter for resync for {TEAM_ID}",
+        mig_0006._alter_table_update_sql(f"WHERE team_id = {TEAM_ID}"),
+        settings={"max_execution_time": 0},
+    )
+    mig_0006._wait_for_mutation_done(QUERY_ID)
+    mig_0006._clear_temporary_tables(QUERY_ID)
+
+
+class Command(BaseCommand):
```

Comment:
Nit: Let's put this as the first thing in the file. Much easier to follow that way.